tslearn/bases.py

```diff
@@ -93,6 +96,8 @@ def _listify(model_params):
 
             if isinstance(param, np.ndarray):
                 model_params[k] = param.tolist()  # for json support
+            elif isinstance(param, list) and isinstance(param[0], np.ndarray):
+                model_params[k] = [p.tolist() for p in param]  # json support
```

Comment:
Why is this part exactly needed? You can call `tolist()` on nD arrays as well:
```python3
>>> x = np.array([[[1]],[[2]],[[3]]])
>>> json.dumps(x.tolist())
```

Reply:
Yes but keras model weights are lists of arrays, not arrays.